javascript/app-discover/src/api/api.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { sleep } from '../utils/Sleep.js'
+import type {
+	DiscoverProgressCallback,
+	DiscoverStartResponse,
+	DiscoverStatusResponse,
+} from './types.js'
+import { DiscoverResponseStatus } from './types.js'
+
+const RUN_CAUSAL_DISCOVERY_BASE_URL = process.env.DISCOVER_API_URL?.endsWith(
+	'/',
+)
+	? process.env.DISCOVER_API_URL
+	: `${process.env.DISCOVER_API_URL || '/api/discover'}/`
+
+export async function fetchDiscoverResult(
+	algorithmName: string,
+	body: string,
+	progressCallback?: DiscoverProgressCallback,
+): Promise<{ result: unknown; taskId: string }> {
```

Comment:
We should add a new type, maybe something like DiscoverResult that looks like: 

```ts
interface DiscoverResult<T>{
 result: T
taskId: string
}
```

Eventually it would be great if types like this could be generated from the service layer automatically.

Reply:
Sure thing